telegram/telegram.py

```diff
@@ -0,0 +1,194 @@
+        static=True,
+        required=True,
+    )
+    outbound_url = ConfigText(
+        'The URL our bot should make requests to',
+        default='https://api.telegram.org/bot', static=True,
+    )
+    inbound_url = ConfigText(
+        'The URL our transport will listen on for Telegram updates',
+        static=True, required=True,
+    )
+
+
+class TelegramTransport(HttpRpcTransport):
+    """
+    Telegram transport for Vumi
+    """
+    transport_type = 'telegram'
+    transport_name = 'telegram_transport'
+
+    CONFIG_CLASS = TelegramTransportConfig
+
+    @classmethod
+    def agent_factory(cls):
+        """For swapping out the Agent for use in tests"""
+        return Agent(reactor)
+
+    @inlineCallbacks
+    def setup_webhook(self):
+        # NOTE: Telegram currently only supports ports 80, 88, 443 and 8443 for
+        #       webhook setup
+        url = '%s/setWebhook' % self.outbound_url.rstrip('/')
+        http_client = HTTPClient(self.agent_factory())
+
+        try:
+            r = yield http_client.post(
+                url=url,
+                data=json.dumps({'url': self.inbound_url}),
+                headers={'Content-Type': ['application/json']}
+                )
+            content = yield r.content()
+            res = json.loads(content)
+            if r.code == http.OK and res['ok']:
+                log.info('Webhook set up on %s' % self.inbound_url)
+            else:
+                log.info('Webhook setup failed: %s' % res['description'])
+
+        # Treat page redirects as errors, since Telegram seems to redirect us
+        # when our bot token is invalid
+        except ResponseFailed as e:
+            log.info(
+                'Webhook setup failed: Invalid token (request redirected)\n%s'
+                % e
+            )
+        # In case we get a response from Telegram that isn't JSON
+        except ValueError as e:
+            log.info(
+                'Webhook setup failed: Unexpected response (expected JSON)\n%s'
+                % e
+            )
+
+    @inlineCallbacks
+    def setup_transport(self):
+        yield super(TelegramTransport, self).setup_transport()
+        config = self.get_static_config()
+        self.outbound_url = '%s%s' % (config.outbound_url.rstrip('/'),
+                                      config.bot_token)
+        self.inbound_url = config.inbound_url
+        self.bot_username = config.bot_username
+
+        yield self.setup_webhook()
+
+    @inlineCallbacks
+    def handle_raw_inbound_message(self, message_id, request):
+        update = json.load(request.content)
+        # TODO: ensure we are not receiving duplicate updates
+
+        # Telegram updates can contain objects other than messages (ignore if
+        # that is the case)
+        if 'message' not in update:
+            log.info('Inbound update does not contain a message')
+            request.finish()
+            return
+        else:
```

Comment:
You can skip the `else:` here if you like since you `return` at the end of `if ...:` (this one is really just personal style and depends a bit on how you like to read this pattern of `check-and-abort`).